FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/IncomingMessageInfo.kt

```diff
@@ -84,6 +91,140 @@ data class IncomingMessageInfo constructor(
+    var body: String? = null
+
+    val folderType = FoldersManager.getFolderType(localFolder)
+    when (messageType) {
+      MessageType.REPLY -> {
+        when (folderType) {
+          FoldersManager.FolderType.SENT,
+          FoldersManager.FolderType.OUTBOX -> {
+            toAddresses.addAll(getTo().map { it.address.lowercase() })
+          }
+
+          else -> {
+            toAddresses.addAll(getReplyToWithoutOwnerAddress().ifEmpty { getTo() }
+              .map { it.address.lowercase() })
+          }
+        }
+      }
+
+      MessageType.REPLY_ALL -> {
+        when (folderType) {
+          FoldersManager.FolderType.SENT, FoldersManager.FolderType.OUTBOX -> {
+            toAddresses.addAll(getTo().map { it.address.lowercase() })
+            ccAddresses.addAll(getCc().map { it.address.lowercase() })
+          }
+
+          else -> {
+            val toRecipients = getReplyToWithoutOwnerAddress().ifEmpty { getTo() }
+            toAddresses.addAll(toRecipients.map { it.address.lowercase() })
+
+            val ccSet = LinkedHashSet<InternetAddress>()
+
+            if (getTo().isNotEmpty()) {
+              ccSet.addAll(getTo().filter { !accountEmail.equals(it.address, ignoreCase = true) })
+
+              for (alias in aliases) {
+                val iterator = ccSet.iterator()
+                while (iterator.hasNext()) {
+                  if (iterator.next().address.equals(alias, ignoreCase = true)) {
+                    iterator.remove()
+                  }
+                }
+              }
+            }
+
+            ccSet.removeAll(toRecipients.toSet())
+            ccSet.addAll(getCc().filter { !accountEmail.equals(it.address, ignoreCase = true) })
+
+            //here we remove the owner address
+            val fromAddress = msgEntity.email
+            val finalCcList = ccSet.filter { !fromAddress.equals(it.address, true) }
+            ccAddresses.addAll(finalCcList.map { it.address.lowercase() })
+          }
+        }
+      }
+
+      MessageType.FORWARD -> {
+        val stringBuilder = StringBuilder()
+        stringBuilder.append(
+          context.getString(
+            R.string.forward_template,
+            getFrom().first().address ?: "",
+            EmailUtil.genForwardedMsgDate(getReceiveDate()),
+            getSubject(),
+            prepareRecipientsLineForForwarding(getTo())
+          )
+        )
+
+        if (getCc().isNotEmpty()) {
+          stringBuilder.append("Cc: ")
+          stringBuilder.append(prepareRecipientsLineForForwarding(getCc()))
+          stringBuilder.append("\n\n")
+        }
+
+        stringBuilder.append("\n\n" + text)
+
+        body = stringBuilder.toString()
+      }
+
+      MessageType.DRAFT -> {
+        toAddresses.addAll(getTo().map { it.address.lowercase() })
+        ccAddresses.addAll(getCc().map { it.address.lowercase() })
+        body = text
+      }
+
+      else -> {}
+    }
+
+    return InitializationData(
```

Comment:
What exactly is `InitializationData` ? What is it used for?

Reply:
It's a data class, that contains info that will be used to init the compose message screen at start up. Like `toResipients` when we create a reply. Or when we use `mailto` schema.